src/engine/source/builder/builders/opBuilderHelperNetInfoAddress.cpp

```diff
@@ -0,0 +1,230 @@
+/* Copyright (C) 2015-2022, Wazuh Inc.
+ * All rights reserved.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+#include "opBuilderHelperNetInfoAddress.hpp"
+
+#include <optional>
+#include <string>
+#include <variant>
+
+#include <baseHelper.hpp>
+#include <kvdb/kvdbManager.hpp>
+#include <wdb/wdb.hpp>
+
+namespace
+{
+
+// TODO: remove pre release or leave it just for testing
+constexpr std::string_view STREAM_SOCK_PATH = "/tmp/testStream.socket";
+
+enum class Name
+{
+    ADDRESS,
+    NETMASK,
+    BROADCAST
+};
+
+constexpr std::string_view getPath(Name field)
+{
+    switch (field)
+    {
+        case Name::ADDRESS: return "/address";
+        case Name::NETMASK: return "/netmask";
+        case Name::BROADCAST: return "/broadcast";
+        default: return "";
+    }
+}
```

Comment:
Wouldn't it be better to use a map for this?

Reply:
I guess that, in this particular case, being just 3 cases, won't make any difference.